### PR TITLE
Remove and deprecate obsolete code and methods

### PR DESCRIPTION
**Pull request by JeethJJ**
With [1], we no longer need to persist default templates with server startup and tenant creation. Hence removing and deprecating the related implementation. 

Addressing : https://github.com/wso2-extensions/identity-event-handler-notification/pull/256#issuecomment-2316811960

[1] https://github.com/wso2-enterprise/asgardeo-product/issues/26100

### CODE REVIEW

File: components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/EmailTemplateManagerImpl.java [hunk 1]
```diff
@@ -422,24 +422,10 @@ public void deleteEmailTemplates(String templateTypeName, String tenantDomain, S
     }
 
     @Override
+    @Deprecated
     public void addDefaultEmailTemplates(String tenantDomain) throws I18nEmailMgtException {
 
-        try {
```
Comment (darshanasbg) [inline components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/EmailTemplateManagerImpl.java:427]: Thinking again, throwing an exception in the deprecated method is not good.. Lets keep the current impl as it is and add a warn log..

File: components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/internal/TenantManagementListener.java [hunk 2]
```diff
@@ -20,16 +20,11 @@
...
-import org.wso2.carbon.email.mgt.exceptions.I18nEmailMgtException;
-import org.wso2.carbon.identity.governance.exceptions.notiification.NotificationTemplateManagerException;
-import org.wso2.carbon.identity.governance.service.notification.NotificationChannels;
-import org.wso2.carbon.identity.governance.service.notification.NotificationTemplateManager;
 import org.wso2.carbon.stratos.common.beans.TenantInfoBean;
 import org.wso2.carbon.stratos.common.exception.StratosException;
 import org.wso2.carbon.stratos.common.listeners.TenantMgtListener;
 
+@Deprecated
 public class TenantManagementListener implements TenantMgtListener {
```
Comment (darshanasbg) [inline components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/internal/TenantManagementListener.java:28]: Lets remove the full class as its not required anymore.

Comment (darshanasbg) [reply components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/internal/TenantManagementListener.java:28]: Also lets avoid registering this as a tenant mgt listener in the I18nMgtServiceComponent

File: components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/EmailTemplateManagerImpl.java (hunk 1 above)
Comment (JeethJJ) [reply components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/EmailTemplateManagerImpl.java:427]: Addressed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/258/commits/2b9424930522b1efcfb51478d134f925b6393352

File: components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/internal/TenantManagementListener.java (hunk 2 above)
Comment (JeethJJ) [reply components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/internal/TenantManagementListener.java]: Addressed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/258/commits/2b9424930522b1efcfb51478d134f925b6393352

**Review by darshanasbg — APPROVED** [review]

**Review by madurangasiriwardena — APPROVED** [review]

**Pull request merged**